### PR TITLE
move jimm interface to a separate file

### PR DESCRIPTION
**Pull request by SimoneDutto**
## Description
I am always annoyed that I can't find the jimm interface, and tha't because it's in a random file.
In this PR i move it to a separate `interface.go` file.
Feel free to suggest another name or close it if 

## Engineering checklist

- [ ] Documentation updated
- [ ] Covered by unit tests
- [ ] Covered by integration tests

### CODE REVIEW

**Review by kian99 — APPROVED** [review]
+1

File: internal/jujuapi/interface.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package jujuapi
```
Comment (alesstimec) [inline internal/jujuapi/interface.go:3]: copyright, please, and a new line after it :)

File: internal/jujuapi/interface.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/canonical/jimm/v3/internal/jimm"
+	"github.com/canonical/jimm/v3/internal/jimm/credentials"
+	"github.com/canonical/jimm/v3/internal/openfga"
+	ofganames "github.com/canonical/jimm/v3/internal/openfga/names"
+	"github.com/canonical/jimm/v3/internal/pubsub"
+	jimmnames "github.com/canonical/jimm/v3/pkg/names"
+	"github.com/go-macaroon-bakery/macaroon-bakery/v3/bakery"
+	"github.com/juju/juju/api/base"
+	jujuparams "github.com/juju/juju/rpc/params"
+	"github.com/juju/names/v5"
```
Comment (alesstimec) [inline internal/jujuapi/interface.go]: imports need sorting

File: internal/jujuapi/interface.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+)
+
+type JIMM interface {
```
Comment (alesstimec) [inline internal/jujuapi/interface.go:26]: godoc, please

File: internal/jujuapi/interface.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	RevokeAuditLogAccess(ctx context.Context, user *openfga.User, targetUserTag names.UserTag) error
+	RevokeCloudAccess(ctx context.Context, user *openfga.User, ct names.CloudTag, ut names.UserTag, access string) error
+	RevokeCloudCredential(ctx context.Context, user *dbmodel.Identity, tag names.CloudCredentialTag, force bool) error
+	RevokeModelAccess(ctx context.Context, user *openfga.User, mt names.ModelTag, ut names.UserTag, access jujuparams.UserAccessPermission) error
+	RevokeOfferAccess(ctx context.Context, user *openfga.User, offerURL string, ut names.UserTag, access jujuparams.OfferAccessPermission) (err error)
+	ToJAASTag(ctx context.Context, tag *ofganames.Tag, resolveUUIDs bool) (string, error)
+	UpdateCloud(ctx context.Context, u *openfga.User, ct names.CloudTag, cloud jujuparams.Cloud) error
+	UpdateCloudCredential(ctx context.Context, u *openfga.User, args jimm.UpdateCloudCredentialArgs) ([]jujuparams.UpdateCredentialModelResult, error)
+	UserLogin(ctx context.Context, identityName string) (*openfga.User, error)
+	ListModels(ctx context.Context, user *openfga.User) ([]base.UserModel, error)
```
Comment (alesstimec) [inline internal/jujuapi/interface.go]: sort lines alphabetically, maybe?

**Review by alesstimec — CHANGES_REQUESTED** [review]

File: internal/jujuapi/interface.go (hunk 1 above)
Comment (SimoneDutto) [reply internal/jujuapi/interface.go:3]: i had the linter disabled..

image

**Review by alesstimec — APPROVED** [review]

**Pull request merged**